sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetTablesOperation.scala

```diff
@@ -124,17 +126,20 @@ private[hive] class SparkGetTablesOperation(
         setState(OperationState.ERROR)
         e match {
           case hiveException: HiveSQLException =>
-            HiveThriftServer2.listener.onStatementError(
-              statementId, hiveException.getMessage, SparkUtils.exceptionString(hiveException))
+            HiveThriftServer2.listener.postLiveListenerBus(SparkListenerStatementError(
+              statementId, hiveException.getMessage, SparkUtils.exceptionString(hiveException),
+              System.currentTimeMillis()))
             throw hiveException
           case _ =>
             val root = ExceptionUtils.getRootCause(e)
-            HiveThriftServer2.listener.onStatementError(
```

Comment:
I mean here

Reply:
I think, without posting the events to `LiveListenerBus`, `EventLoggingListner` won't be able to catch the events, and hence `Historyserver` won't populate the page

Reply:
> I think, without posting the events to `LiveListenerBus`, `EventLoggingListner` won't be able to catch the events, and hence `Historyserver` won't populate the page

I know the whole process. I mean not change API in Operation classes. 
Keep change in ThriftServerListener and post it to ListenerBus

Reply:
Thanks, let me try that

Reply:
:+1: to @AngersZhuuuu suggestion. To avoid changes in the operation classes, you could call keep calling the onFooBar functions directly there, and then `if (live) { postLiveListenerBus(createEvent) }` from there.

Reply:
> 👍 to @AngersZhuuuu suggestion. To avoid changes in the operation classes, you could call keep calling the onFooBar functions directly there, and then if (live) { postLiveListenerBus(createEvent) } from there.

I have updated based on the above comment. Now, the operation classes won't have any changes.